### DISCUSSION TITLE
Passive URL Resolution

### DISCUSSION THREAD

**Discussion by OneYellowLemon** [QuorumDMS/ftp-srv#272] [Q&A] [answered]
Hello! I am trying to have an FTP server work over both local IP and public IP, and I am trying to get the passive URL to work this way. As per the README, this code is provided. 
```js
const {Netmask} = require('netmask');
const networks = {
 '$GATEWAY_IP/32': `${public_ip}`, 
 '10.0.0.0/8'    : `${lan_ip}`
} 

const resolverFunction = (address) => {
   for (const network in networks) {
       try {
           const mask = new Netmask(network);
           if (mask.contains(address)) return networks[network];
       }
       catch (err) {
           logger.error('Error checking source network', err);
       }
   }
   return '127.0.0.1';
}

new FtpSrv({pasv_url: resolverFunction});
```
However, I can't figure out how I would make this work without assuming what gateway IP and netmask people will have on their LANs. Is anyone able to help me?

**Comment by antonilol** [comment, marked as answer]
Get the gateway ip address with either a node js library or a system command

**Reply by OneYellowLemon** [reply]
How would I get the network submask? I believe I would have to parse command line utils, is there any other way to do it?

**Reply by antonilol** [reply]
with the command (replace `wlp1s0` with your network interface name) `ifconfig wlp1s0 | grep netmask | awk '{ printf $4 }'`

**Reply by antonilol** [reply]
oh and also possible with command parsing: https://net-raft.com/Questions/1376/how-to-get-client-netmask-using-nodejs-/1376
i tested it and it works for me

**Reply by matt-forster** [reply]
Thanks for the answer @antonilol!
